### ISSUE TITLE
DRM not working 

### ISSUE THREAD

**Issue by victor-50** [apprenticeharper/DeDRM_tools#2282]
November 12 DRM still worked ok. Three books epub books I added yesterday seemed to be decrypted, but couldn't be read.

calibre, version 7.1.0
FOUT: Boek laden mislukt:  Donna Leon\A Sea of Troubles (272)\A Sea of Troubles - Donna Leon.epub mislukt. Klik op "Toon details" voor meer info.

Failed to convert book: Donna Leon\A Sea of Troubles (272)\A Sea of Troubles - Donna Leon.epub with error:
DeDRM v7.2.1: Trying to decrypt A Sea of Troubles - Donna Leon.epub
DeDRM v7.2.1: Verifying zip archive integrity
DeDRM v7.2.1: A Sea of Troubles - Donna Leon.epub is a secure Adobe Adept ePub
DeDRM v7.2.1: Trying Encryption key default_key
Could not decrypt 7sf9cq__.epub because of an exception:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.ineptepub", line 425, in decryptBook
  File "calibre_plugins.dedrm.ineptepub", line 343, in __init__
  File "calibre_plugins.dedrm.ineptepub", line 298, in __init__
  File "Crypto\Cipher\AES.py", line 228, in new
  File "Crypto\Cipher\__init__.py", line 79, in _create_cipher
  File "Crypto\Cipher\_mode_cbc.py", line 274, in _create_cbc_cipher
  File "Crypto\Cipher\AES.py", line 90, in _create_base_cipher
ValueError: Incorrect AES key length (0 bytes)

DeDRM v7.2.1: Failed to decrypt with key default_key after 0.2 seconds
DeDRM v7.2.1: Looking for new default Adobe Digital Editions Keys after 0.2 seconds
Found 1 keys
DeDRM v7.2.1: Trying a new default key
DeDRM v7.2.1: Exception when decrypting after 0.2 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 384, in ePubDecrypt
  File "calibre_plugins.dedrm.ineptepub", line 399, in decryptBook
  File "calibre_plugins.dedrm.ineptepub", line 306, in __init__
  File "calibre_plugins.dedrm.ineptepub", line 306, in <listcomp>
  File "calibre_plugins.dedrm.ineptepub", line 283, in getChild
  File "calibre_plugins.dedrm.ineptepub", line 220, in get
calibre_plugins.dedrm.ineptepub._load_crypto_pycrypto.<locals>.ASN1Error: Error decoding ASN.1
DeDRM v7.2.1: Failed to decrypt with new default key after 0.2 seconds
DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 187, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 644, in run
  File "calibre_plugins.dedrm.__init__", line 415, in ePubDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
InputFormatPlugin: EPUB Input running
on Donna Leon\A Sea of Troubles (272)\A Sea of Troubles - Donna Leon.epub
Failed to run pipe worker with command: from calibre.srv.render_book import viewer_main; viewer_main()
Traceback (most recent call last):
  File "runpy.py", line 198, in _run_module_as_main
  File "runpy.py", line 88, in _run_code
  File "site.py", line 83, in <module>
  File "site.py", line 78, in main
  File "site.py", line 50, in run_entry_point
  File "calibre\utils\ipc\worker.py", line 196, in main
  File "<string>", line 1, in <module>
  File "calibre\srv\render_book.py", line 952, in viewer_main
  File "calibre\srv\render_book.py", line 943, in render_for_viewer
  File "calibre\srv\render_book.py", line 920, in render
  File "calibre\ebooks\oeb\iterator\book.py", line 62, in extract_book
  File "calibre\customize\conversion.py", line 242, in __call__
  File "calibre\ebooks\conversion\plugins\epub_input.py", line 287, in convert
calibre.ebooks.DRMError: A Sea of Troubles - Donna Leon.epub
Exception ignored in: <function NativeFunction.__del__ at 0x0000025C2EF64AE0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function NativeFunction.__del__ at 0x0000025C2EF64AE0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is not callable

**Comment by ElleKayEm** [comment]
Try DeDRM 10.0.9: https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.9

**Comment by victor-50** [comment]
I tried. Now I get a message that the book is protected by DRM! Seems they changed it again.

**Comment by ElleKayEm** [comment]
Let's see a new log.

**Comment by victor-50** [comment]
This time just the message, no log. Maybe the log above contains enough info. I could send you the book by mail.

**Comment by victor-50** [comment]
As my PC didn't work I did this on my friends. I now see that there Adobe was updated to 4.5. I think that's the problem. Keep you posted if I can resolve it.

**Comment by victor-50** [comment]
Sorry again. Turns out to be version 2.0 after all. There was an old link on the desktop to 4.5

**Comment by victor-50** [comment]
I just tried epubor_ultimate. It decrypted (trial, so only 20%).

**Comment by ElleKayEm** [comment]
Epubor use the code from this project.  So if that worked, this will too.  Here's how to get the log: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by victor-50** [comment]
I hope it is ok to paste the log here: 
ApplicationPaletteChange event ignored
Using calibre Qt style: True
ApplicationPaletteChange event received
calibre Foutopsporingslog
calibre 7.1  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: nl
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 5, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 5, 0) && Category Tags (0, 2, 8) && Count Pages (1, 13, 4) && KFX metadata reader (from KFX Input) (2, 5, 0) && KFX Input (2, 5, 0) && Open With (1, 8, 1)
calibre 7.1  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: nl
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 5, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 5, 0) && Category Tags (0, 2, 8) && Count Pages (1, 13, 4) && KFX metadata reader (from KFX Input) (2, 5, 0) && KFX Input (2, 5, 0) && Open With (1, 8, 1)
QPA platform: windows
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 92.53889943074003 x 92.67567567567568
[0.00] Starting up...
[0.00] Showing splash screen...
[0.17] splash screen shown
[0.17] Initializing db...
[0.44] db initialized
[0.44] Constructing main UI...
Category Tags: User Categories: Import User Catetories: running on_modules_update()
Category Tags: User Categories: Export User Catetories: running on_modules_update()
Key 'B' for shortcut Doorblader annotaties is already used by Boeken bijschaven, ignoring
Key 'B' for shortcut Doorblader annotaties is already used by Boeken bijschaven, ignoring
[1.55] main UI initialized...
[1.55] Hiding splash screen
Starting QuickView
DeDRM v10.0.9: Trying to decrypt The Jewels of Paradise.epub
DeDRM v10.0.9: Verifying zip archive integrity
DeDRM v10.0.9: The Jewels of Paradise.epub is a secure Adobe Adept ePub for UUID cd02f12a-3816-45ae-993d-e754949a02a0
DeDRM v10.0.9: Trying Encryption key default_key
p3_zjexu.epub is a secure Adobe Adept ePub.
Could not decrypt p3_zjexu.epub because of an exception:
Traceback (most recent call last):
  File "C:\Users\valk\AppData\Roaming\calibre\plugins\DeDRM.zip\ineptepub.py", line 275, in decryptBook
  File "C:\Users\valk\AppData\Roaming\calibre\plugins\DeDRM.zip\ineptepub.py", line 91, in __init__
  File "Crypto\Cipher\AES.py", line 228, in new
  File "Crypto\Cipher\__init__.py", line 79, in _create_cipher
  File "Crypto\Cipher\_mode_cbc.py", line 274, in _create_cbc_cipher
  File "Crypto\Cipher\AES.py", line 90, in _create_base_cipher
ValueError: Incorrect AES key length (0 bytes)

DeDRM v10.0.9: Failed to decrypt with key default_key after 0.2 seconds
DeDRM v10.0.9: Looking for new default Adobe Digital Editions Keys after 0.2 seconds
Found 1 keys
DeDRM v10.0.9: Trying a new default key
p3_zjexu.epub is a secure Adobe Adept ePub.
Could not decrypt p3_zjexu.epub because of an exception:
Traceback (most recent call last):
  File "C:\Users\valk\AppData\Roaming\calibre\plugins\DeDRM.zip\ineptepub.py", line 253, in decryptBook
  File "Crypto\PublicKey\RSA.py", line 822, in import_key
ValueError: RSA key format is not supported

DeDRM v10.0.9: Failed to decrypt with new default key after 0.3 seconds
DeDRM v10.0.9: Ultimately failed to decrypt after 0.3 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 187, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 1041, in run
  File "calibre_plugins.dedrm.__init__", line 639, in ePubDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v10.0.9: Ultimately failed to decrypt after 0.3 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md
[23.55] splash screen hidden
[23.55] Started up in 23.55 seconds with 0 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\x9e\x03\x95\x95\x9f|8#\x06\xa4\xaa\xe2\x92\r~\x0b\xc1V\x1c\x89~D\xd6; ,\xa6\xa8\xdeo\x8d2', b'\\\\HVW-0\\Install\\The Jewels of Paradise.epub']
Added The Jewels of Paradise to db in: 0.2
Added 1 books in 0.7 seconds

**Comment by ElleKayEm** [comment]
Delete the saved Adobe Key in DeDRM's customization.  Add your book again letting DeDRM retrieve the key automatically.

**Comment by victor-50** [comment]
I deleted the first key in the sjon file. That didn't help. I deleted all DeDRM files in AppData\Roaming\calibre\plugins\DeDRM.zip and reïnstalled DeDRM. Didn't work either. The sjon now looks like this:
{
  "adeptkeys": {},
  "adobe_pdf_passphrases": [],
  "androidkeys": {},
  "bandnkeys": {},
  "ereaderkeys": {},
  "kindlekeys": {},
  "lcp_passphrases": [],
  "pids": [],
  "serials": []
}
I think and hope I have a backup, because It looks possible that I have to import books again I think
I wasn't able to switch back to normal (non-logging) operation of Calibre. I tried to point to the library, but it said it was already the library.

**Comment by ElleKayEm** [comment]
Wow, that's not what I suggested doing.  And I don't know how you've managed this.

If ADE (where you downloaded the book) and calibre/DeDRM are on the same computer and user, then DeDRM will automatically retrieve your key and remove the DRM when you add the epub to calibre.

**Comment by victor-50** [comment]
Hi, I owe you an explanation. I was having trouble transiting to a new (W11) system and had my old SSD in the new PC. ADE asked to register anew, but I was not ready for that. So I downloaded on the ADE of my partner. I thought I had everything installed the same way. When DeDRM didn't work I copied the downloaded epubs to my machine and tried there, not realizing/knowing that DeDRM only works for files on ADE on the same machine.  After your last comment I checked the PC of my partner and found out DeDRM was not installed. When I installed it, it worked. I managed to restore the sjon file. Thanks for your help!
